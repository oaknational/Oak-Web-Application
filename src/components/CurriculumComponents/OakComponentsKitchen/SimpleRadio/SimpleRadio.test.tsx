import { RadioGroup, RadioButton } from ".";

import renderWithTheme from "@/__tests__/__helpers__/renderWithTheme";

describe("Fieldset", () => {
  test("render / none", () => {
    const { baseElement } = renderWithTheme(
      <RadioGroup name="testing" onChange={(e) => console.log(e)} value={null}>
        <RadioButton value="one">one</RadioButton>
        <RadioButton value="two">two</RadioButton>
      </RadioGroup>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  test("render / selected", () => {
    const { baseElement } = renderWithTheme(
      <RadioGroup name="testing" onChange={(e) => console.log(e)} value="one">
        <RadioButton value="one">one</RadioButton>
        <RadioButton value="two">two</RadioButton>
      </RadioGroup>,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
