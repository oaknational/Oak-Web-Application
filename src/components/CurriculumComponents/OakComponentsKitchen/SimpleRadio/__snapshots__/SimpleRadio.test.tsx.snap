// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fieldset render / none 1`] = `
<body>
  <div>
    <label
      style="display: flex; align-items: center; margin-bottom: 12px; opacity: 1;"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="testing"
          type="radio"
          value="one"
        />
      </div>
      <span
        class="sc-3299cdd2-0 iHfbvZ"
      />
      one
    </label>
    <label
      style="display: flex; align-items: center; margin-bottom: 12px; opacity: 1;"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="testing"
          type="radio"
          value="two"
        />
      </div>
      <span
        class="sc-3299cdd2-0 iHfbvZ"
      />
      two
    </label>
  </div>
</body>
`;

exports[`Fieldset render / selected 1`] = `
<body>
  <div>
    <label
      style="display: flex; align-items: center; margin-bottom: 12px; opacity: 1;"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          checked=""
          name="testing"
          type="radio"
          value="one"
        />
      </div>
      <span
        class="sc-3299cdd2-0 KzrSU"
      />
      one
    </label>
    <label
      style="display: flex; align-items: center; margin-bottom: 12px; opacity: 1;"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="testing"
          type="radio"
          value="two"
        />
      </div>
      <span
        class="sc-3299cdd2-0 iHfbvZ"
      />
      two
    </label>
  </div>
</body>
`;
